src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -614,6 +199,37 @@ public void setAbortThreshold(int abortThreshold) {
         this.abortThreshold = abortThreshold == 0 ? -1 : abortThreshold;
     }
 
+    /**
+     * Tells the validator to use the specified number of threads during validation
+     *
+     * @param threadCount number of threads to use during validation
+     */
+    public void setThreadCount(int threadCount) {
+        if (threadCount < 1) {
+            throw new IllegalArgumentException("Number of threads must be greater than or equal to 1.");
+        }
+        if (threadCount > 1 && (executor == null || executor.getPoolSize() != threadCount)) {
+            executor = new ThreadPoolExecutor(threadCount, threadCount,0L, TimeUnit.MILLISECONDS,
+                    new LinkedBlockingQueue<Runnable>());
+        } else if (threadCount == 1 && executor != null) {
+            executor.shutdown();
+            executor = null;
+        }
+    }
+
+    /**
+     * Return the current executor if one exists, null o/w
+     * @return the current executor if one exists, null o/w
+     */
+    public ExecutorService getExecutor() {
+        return executor;
+    }
+
+    //TODO: remove this
+    public List<Future<ThreadedValidationEngine.ValidationMetadata>> getValidationMetadata() {
```

Comment:
This can't be removed until we get rid of the performance test. Let me know what you think about removing it now. Do you think we could use the futures for the performance monitor?

Reply:
Gotcha-- just change comment to say something like remove this when `ValidatorPerformanceTest` is removed.  Although I'm not opposed to keeping the test (and therefore this method).

Reply:
Fixed